app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,24 @@
+class CommentsController < ApplicationController
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.build(comment_params.merge(post: @post))
+
+    if @comment.save
+      @post.update(comments_counter: @post.comments.count)
+      redirect_to user_post_path(@post.author, @post)
+    else
+      redirect_to user_post_path(@post.author, @post), alert: 'Unable to add the comment.'
+    end
+  end
```

Comment:
- [ ] please make sure the comment is created on behalf of the current logged in user.

Reply:
- Well resolved